x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -208,32 +193,48 @@ export const createInventoryMetricThresholdExecutor = (libs: InfraBackendLibs) =
           libs.basePath
         );
         scheduledActionsCount++;
-        alert.scheduleActions(
-          /**
-           * TODO: We're lying to the compiler here as explicitly  calling `scheduleActions` on
-           * the RecoveredActionGroup isn't allowed
-           */
-          actionGroupId as unknown as InventoryMetricThresholdAllowedActionGroups,
-          {
-            group,
-            alertState: stateToAlertMessage[nextState],
-            reason,
-            timestamp: startedAt.toISOString(),
-            viewInAppUrl,
-            value: mapToConditionsLookup(results, (result) =>
-              formatMetric(result[group].metric, result[group].currentValue)
-            ),
-            threshold: mapToConditionsLookup(criteria, (c) => c.threshold),
-            metric: mapToConditionsLookup(criteria, (c) => c.metric),
-          }
-        );
+        const context = {
+          group,
+          alertState: stateToAlertMessage[nextState],
+          reason,
+          timestamp: startedAt.toISOString(),
+          viewInAppUrl,
+          value: mapToConditionsLookup(results, (result) =>
+            formatMetric(result[group].metric, result[group].currentValue)
+          ),
+          threshold: mapToConditionsLookup(criteria, (c) => c.threshold),
+          metric: mapToConditionsLookup(criteria, (c) => c.metric),
+        };
+        alert.scheduleActions(actionGroupId, context);
       }
     }
+
+    const { getRecoveredAlerts } = services.alertFactory.done();
+    const recoveredAlerts = getRecoveredAlerts();
+    for (const alert of recoveredAlerts) {
+      const recoveredAlertId = alert.getId();
+      const indexedStartedDate = getAlertStartedDate(recoveredAlertId) ?? startedAt.toISOString();
+      const viewInAppUrl = getViewInAppUrlInventory(
+        criteria,
+        nodeType,
+        indexedStartedDate,
+        libs.basePath
+      );
+      const context = {
```

Comment:
@simianhacker Do we plan to add later on the `value`, `reason` and `alertState` in the context of recovered alerts similar to how we do above for active alerts?

Reply:
When you set the `reason` message, it's ignored by the alerting system; it just recovers using the old `reason` message. 

For the `value`, we would need to return the data for recovered "entities" along with the ones that matched the threshold. This would negate all the optimization work we did to push the evaluations into ES. I've talked to @vinaychandrasekhar about both of these issues.